### PR TITLE
Add InsideTheBackStack sample

### PR DESCRIPTION
**Pull request by vladcipariu91**
## Description

Added a new sample app to show how the back stack works, this was adapter from Challenge 2 from DroidCon London

It's the last item in the list:

appyx-sample

## Check list

- [x] I have updated `CHANGELOG.md` if required. 
- [x] I have updated documentation if required. -

### CODE REVIEW

File: samples/app/src/main/kotlin/com/bumble/appyx/app/node/backstack/app/custombackstack/CustomBackStack.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package com.bumble.appyx.app.node.backstack.app.custombackstack
+
+import com.bumble.appyx.core.navigation.BaseNavModel
+import com.bumble.appyx.core.navigation.NavElements
+import com.bumble.appyx.core.navigation.NavKey
+import com.bumble.appyx.core.navigation.Operation
+import com.bumble.appyx.core.state.SavedStateMap
+
+class CustomBackStack<NavTarget : Any>(
```
Comment (zsoltk) [inline samples/app/src/main/kotlin/com/bumble/appyx/app/node/backstack/app/custombackstack/CustomBackStack.kt]: Can we simply use the standard BackStack instead?

**Comment by zsoltk** [comment]
We probably don't need to add the `logo.png`, there should be one in this project already.

File: samples/app/src/main/kotlin/com/bumble/appyx/app/node/backstack/app/custombackstack/CustomBackStack.kt (hunk 1 above)
Comment (vladcipariu91) [reply samples/app/src/main/kotlin/com/bumble/appyx/app/node/backstack/app/custombackstack/CustomBackStack.kt]: we'll lose the animation if we use the regular BackStack

**Comment by vladcipariu91** [comment]
> Details

I can remove the Logo

**Review by zsoltk — APPROVED** [review]

**Pull request merged**